src/webgl/p5.Shader.js

```diff
@@ -471,17 +472,37 @@ p5.Shader.prototype.isStrokeShader = function() {
  * @private
  */
 p5.Shader.prototype.enableAttrib = function(
-  loc,
+  attr,
   size,
   type,
   normalized,
   stride,
   offset
 ) {
-  var gl = this._renderer.GL;
-  if (loc !== -1) {
-    gl.enableVertexAttribArray(loc);
-    gl.vertexAttribPointer(loc, size, type, normalized, stride, offset);
+  if (attr) {
+    if (
+      typeof IS_MINIFIED === 'undefined' &&
+      this.attributes[attr.name] !== attr
+    ) {
+      console.error(
+        'The attribute "' +
+          attr.name +
+          '"passed to enableAttrib does not belong to this shader.'
```

Comment:
Maybe log `this.attributes` in here somewhere, or something else to help identify the shader. The stack trace may be enough but the 'this shader' seems a bit vague.

Recently @lmccart mentioned that p5 tries not to use `console.error` in a review of a different PR. I understand that this is a breaking condition without this check, but maybe `console.log` or at most `console.warn`.

Reply:
Ah yeah I'll fix this